cloud/src/meta-service/meta_service_http.cpp

```diff
@@ -331,6 +338,128 @@ static HttpResponse process_alter_iam(MetaServiceImpl* service, brpc::Controller
+                                           qps_limit_str, rpc_name, instance_id));
+    };
+
+    static auto parse_qps_limit =
+            [](const std::string& qps_limit_str) -> std::variant<int64_t, HttpResponse> {
+        DCHECK(!qps_limit_str.empty());
+        int64_t qps_limit = -1;
+        try {
+            qps_limit = std::stoll(qps_limit_str);
+        } catch (const std::exception& ex) {
+            return http_json_reply(
+                    MetaServiceCode::INVALID_ARGUMENT,
+                    fmt::format("param `qps_limit` is not a legal int64 type:{}", ex.what()));
+        }
+        if (qps_limit < 0) {
+            return http_json_reply(MetaServiceCode::INVALID_ARGUMENT,
+                                   "`qps_limit` should not be less than 0");
+        }
+        return qps_limit;
+    };
+
+    auto process_set_qps_limit = [&](std::function<bool(int64_t)> cb) -> HttpResponse {
+        return std::visit(
+                [&](auto&& parse_result) {
+                    using T = std::decay_t<decltype(parse_result)>;
+                    if constexpr (std::is_same_v<T, HttpResponse>) {
+                        return parse_result;
+                    } else {
+                        if (cb(parse_result)) {
+                            return http_json_reply(MetaServiceCode::OK,
+                                                   "sucess to adjust rate limit");
+                        }
+                        return http_json_reply(
+                                MetaServiceCode::INVALID_ARGUMENT,
+                                fmt::format("failed to adjust rate limit for qps_limit={}, "
+                                            "rpc_name={}, instance_id={}, plz ensure correct "
+                                            "rpc/instance name",
+                                            qps_limit_str, rpc_name, instance_id));
+                    }
+                },
+                parse_qps_limit(qps_limit_str));
+    };
+
+    auto set_global_qps_limit = [process_set_qps_limit, service]() {
+        return process_set_qps_limit([service](int64_t qps_limit) {
+            return service->rate_limiter()->set_rate_limit(qps_limit);
+        });
+    };
+
+    auto set_rpc_qps_limit = [&]() {
+        return process_set_qps_limit([&](int64_t qps_limit) {
+            return service->rate_limiter()->set_rate_limit(qps_limit, rpc_name);
+        });
+    };
+
+    auto set_instance_qps_limit = [&]() {
+        return process_set_qps_limit([&](int64_t qps_limit) {
+            return service->rate_limiter()->set_instance_rate_limit(qps_limit, instance_id);
+        });
+    };
+
+    auto set_instance_rpc_qps_limit = [&]() {
+        return process_set_qps_limit([&](int64_t qps_limit) {
+            return service->rate_limiter()->set_rate_limit(qps_limit, rpc_name, instance_id);
+        });
+    };
+
+    // for 8 element in true table of params, register processor cb
```

Comment:
this part is confusing

Reply:
Maybe I'll add some comments, just register different callback for different params pattern, and the "abuse" of lamda are just meant to adapt for this.